common/fitparser.h

```diff
@@ -0,0 +1,100 @@
+/* fitparser.h
+
+Copyright (c) 2022, Nikolaj Schlej. All rights reserved.
+
+This program and the accompanying materials
+are licensed and made available under the terms and conditions of the BSD License
+which accompanies this distribution.  The full text of the license may be found at
+http://opensource.org/licenses/bsd-license.php.
+
+THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+
+*/
+
+#ifndef FITPARSER_H
+#define FITPARSER_H
+
+#include <vector>
+
+#include "basetypes.h"
+#include "ustring.h"
+#include "ubytearray.h"
+#include "treemodel.h"
+#include "intel_fit.h"
+#include "intel_microcode.h"
+#include "ffsparser.h"
+
+class FfsParser;
+
+#ifdef U_ENABLE_FIT_PARSING_SUPPORT
+class FitParser
+{
+public:
+    // Default constructor and destructor
+    FitParser(TreeModel* treeModel, FfsParser* parser) : model(treeModel), ffsParser(parser),
+        bgAcmFound(false), bgKeyManifestFound(false), bgBootPolicyFound(false) {}
+    ~FitParser() {}
+
+    // Returns messages
+    std::vector<std::pair<UString, UModelIndex> > getMessages() const { return messagesVector; }
+    // Clears messages
+    void clearMessages() { messagesVector.clear(); }
+
+    // Obtain parsed FIT table
+    std::vector<std::pair<std::vector<UString>, UModelIndex> > getFitTable() const { return fitTable; }
+    
+    // Obtain security info
+    UString getSecurityInfo() const { return securityInfo; }
+    
+    // FIT parsing
+    USTATUS parseFit(const UModelIndex & index);
+        
+private:
+    TreeModel *model;
```

Comment:
To be honest, maybe zero these inline for safety? Like TreeModel * model {nullptr}; and so on?

Reply:
They are initialised by constructor init list, and if I also do it here, analysers will berate me for it. SonarCloud already does. :)

Reply:
Now that we have C++11, perhaps it is the time to get rid of constructor lists? They are legacy and fugly. Probably better in a separate commit.

Reply:
Separate commit for sure.